kwok/cloudprovider/helpers.go

```diff
@@ -30,78 +32,102 @@ import (
 	"sigs.k8s.io/karpenter/pkg/scheduling"
 )
 
+var (
+	// AWS uses (family).(size) format
+	awsRegexp = regexp.MustCompile(`^\w+\.(nano|micro|small|medium|large|\d*xlarge|metal)$`)
+
+	familyDelim = regexp.MustCompile(`[.-]`)
+)
+
 type InstanceTypeOptions struct {
-	Name               string
-	Offerings          cloudprovider.Offerings
-	Architecture       string
-	OperatingSystems   sets.Set[string]
-	Resources          v1.ResourceList
-	InstanceTypeLabels map[string]string
+	Name             string                  `json:"name"`
+	Offerings        cloudprovider.Offerings `json:"offerings"`
+	Architecture     string                  `json:"architecture"`
+	OperatingSystems sets.Set[string]        `json:"operatingSystems"`
```

Comment:
Can we change this to a list of strings rather than a set? Should make the Json blob a bit easier to reason about. 
```suggestion
	OperatingSystems []string        `json:"operatingSystems"`
```

Reply:
I converted it to a v1.OSName.